client/consensus/manual-seal/src/heartbeat_stream.rs

```diff
@@ -0,0 +1,100 @@
+// Library shortcuts
+
+use std::{pin::Pin};
+use futures::{
+	prelude::*,
+	task::{Context, Poll},
+};
+use tokio::time::{Duration, Instant, Delay};
+
+use crate::rpc::{EngineCommand};
+
+// ---
+// Constant Definition
+
+const DEFAULT_TIMEOUT: u64 = 30;
+const DEFAULT_MIN_BLOCKTIME: u64 = 1;
+const DEFAULT_FINALIZE: bool = false;
+
+// ---
+pub struct HeartbeatOptions {
+	// the amount of time passed that a new heartbeat block will be generated, in sec.
+	pub timeout: u64,
+	// the minimum amount of time to pass before generating another block, in sec.
+	pub min_blocktime: u64,
+	// whether the generated heartbeat block is finalized
+	pub finalize: bool,
+}
+
+impl Default for HeartbeatOptions {
+	fn default() -> Self {
+		Self {
+			timeout: DEFAULT_TIMEOUT,
+			min_blocktime: DEFAULT_MIN_BLOCKTIME,
+			finalize: DEFAULT_FINALIZE,
+		}
+	}
+}
+
+pub struct HeartbeatStream<Hash> {
+	pool_stream: Box<dyn Stream<Item = EngineCommand<Hash>> + Unpin + Send>,
+	delay: Delay,
```

Comment:
What is this delay field for? Is it different than the heartbeat period and min block period from the `HeartbeatOptions` struct?